server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -2010,7 +2024,7 @@ public IndexMetadata build() {
                 mapping,
                 aliases.build(),
                 newCustomMetadata,
-                Map.ofEntries(denseInSyncAllocationIds),
+                denseInSyncAllocationIds,
```

Comment:
I prefer to make immutable here rather than in the constructor, that makes the choice of how to do that local rather than adding a wrapper and sometimes maybe multiple wrappers.

Reply:
Sounds good, I moved the Collections.unmodifiableList from the constructor to the appropriate places where the constructor is called. This should avoid multiple wrappers now.